go/app/infra.go

```diff
@@ -28,44 +27,62 @@ type ItemRepository interface {
 	Insert(ctx context.Context, item *Item) error
 	GetAllItem(ctx context.Context) ([]Item, error)
 	GetItemById(ctx context.Context, itemId string) (Item, error)
+	SearchItemsByKeyword(ctx context.Context, keyword string) ([]Item, error)
 }
 
 // itemRepository is an implementation of ItemRepository
 type itemRepository struct {
-	// fileName is the path to the JSON file storing items.
-	fileName string
+	db *sql.DB
 }
 
 // NewItemRepository creates a new itemRepository.
-func NewItemRepository() ItemRepository {
-	return &itemRepository{fileName: "items.json"}
+func NewItemRepository(db *sql.DB) ItemRepository {
+	return &itemRepository{db: db}
 }
 
-// Insert inserts an item into the repository.
-func (i *itemRepository) Insert(ctx context.Context, item *Item) error {
-	// STEP 4-2: add an implementation to store an item
-	items, err := i.GetAllItem(ctx)
-	if err != nil {
-		return err
-	}
+func initDB(db *sql.DB) error {
+	sqlFile := "db/items.sql"
```

Comment:
[nits]
不変な変数はconstで宣言すると良いとされています
静的な変数であることが明示的に分かりますし、メモリ的にも毎回確保されなくなるので

```suggestion
	const sqlFile = "db/items.sql"
```